components/brave_rewards/resources/extension/brave_rewards/content_scripts/twitter.ts

```diff
@@ -7,7 +7,62 @@ import { getMessage } from '../background/api/locale_api'
 
 let timeout: any = null
 
-const getTweetMetaData = (tweet: Element): RewardsTip.TweetMetaData | null => {
+const getTwitterAPICredentials = () => {
+  const msg = { type: 'getTwitterAPICredentials' }
+  return new Promise(resolve => chrome.runtime.sendMessage(msg, resolve))
+}
+
+const getTweetDetails = async (tweetId: string) => {
+  const credentialHeaders = await getTwitterAPICredentials()
+  const url = new URL('https://api.twitter.com/1.1/statuses/show.json')
+  url.searchParams.append('id', tweetId)
+  const response = await fetch(url.toString(), {
+    credentials: 'include',
+    headers: {
+      ...credentialHeaders
+    },
+    referrerPolicy: 'no-referrer-when-downgrade',
+    method: 'GET',
+    mode: 'cors',
+    redirect: 'follow'
+  })
+  return response.json()
+}
+
+const getTweetMetaData = (tweet: Element) => {
+  if (!tweet) {
+    return Promise.reject(null)
+  }
+
+  const status = tweet.querySelector("a[href*='/status/']") as HTMLAnchorElement
+  if (!status || !status.href) {
+    return Promise.reject(null)
+  }
+
+  const tweetIdMatches = status.href.match(/status\/(\d+)/)
+  if (!tweetIdMatches || tweetIdMatches.length < 2) {
+    return Promise.reject(null)
+  }
+
+  const tweetId = tweetIdMatches[1]
+
+  return getTweetDetails(tweetId)
+    .then(tweetDetails => {
+      return {
+        name: tweetDetails.user.name,
+        screenName: tweetDetails.user.screen_name,
+        userId: tweetDetails.user.id_str,
+        tweetId: tweetId,
+        tweetTimestamp: Date.parse(tweetDetails.created_at) / 1000,
+        tweetText: tweetDetails.text
+      }
+    })
+    .catch(error => {
```

Comment:
We should re-throw this catch, since otherwise the caller won't know nothing wrong happened. Instead the promise will get resolved with `null`. I can add that in to the rebase @emerick